internal/backup_mark.go

```diff
@@ -0,0 +1,81 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/wal-g/wal-g/internal/storages/storage"
+	"github.com/wal-g/wal-g/internal/tracelog"
+	"github.com/wal-g/wal-g/utility"
+)
+
+// MarkSelfAndPreviousBackupsPermanent marks all previous related backups
+// permanent, including itself, any previous delta backups and initial full
+// backup
+// TODO: unit tests
+func MarkSelfAndPreviousBackupsPermanent(uploader *Uploader, baseBackupFolder storage.Folder, backupName string) error {
+	toUpload := []UploadObject{}
+
+	// mark self if impermanent
```

Comment:
'if -> as'?

Reply:
not really, i wanted to mean "mark if not already permanent" to avoid uploading objects unnecessarily. This is also useful when we eventually add the `makeBackupPermanent` command.

Reply:
Yes, I misread this, it's ok.